local/network.go

```diff
@@ -448,11 +445,15 @@ func (ln *localNetwork) Healthy(ctx context.Context) chan error {
 		return healthyChan
 	}
 
-	nodes := make([]*localNode, 0, len(ln.nodes))
-	for _, node := range ln.nodes {
-		nodes = append(nodes, node)
-	}
 	go func() {
+		ln.lock.RLock()
```

Comment:
this works, but it keeps the network locked during the health check. do we want that?
if the answer is no, then a more complex health is needed, that restart if the list of nodes changes.

Reply:
Maybe someone else should have a say at this one, but honestly I think it's not too bad to lock during the health check. When a DB is queried you also read-lock otherwise you might get stale results...

Reply:
The problem here is that the health check itself waits, so we should probably fix that - but possibly in another PR

Reply:
As per solving the current race condition this work. LGTM would approve but let's please add a TODO comment for it, probably mentioning possible future implementation